### PR TITLE
Impl sceNpBasicSendMessage

### PR DESCRIPTION
**Pull request by RipleyTom**
Also fixes a bug where the data associated with the message was never given out to the application.

### CODE REVIEW

File: rpcs3/Emu/NP/np_handler.cpp [hunk 1]
```diff
@@ -1109,15 +1115,15 @@ namespace np
 								event = (msg->second.msgFeatures & SCE_NP_BASIC_MESSAGE_FEATURES_BOOTABLE) ? SCE_NP_BASIC_EVENT_INCOMING_BOOTABLE_CUSTOM_DATA_MESSAGE : SCE_NP_BASIC_EVENT_INCOMING_CUSTOM_DATA_MESSAGE;
 								break;
 							case SCE_NP_BASIC_MESSAGE_MAIN_TYPE_GENERAL:
-							case SCE_NP_BASIC_MESSAGE_MAIN_TYPE_ADD_FRIEND:
-							case SCE_NP_BASIC_MESSAGE_MAIN_TYPE_URL_ATTACHMENT:
 								event = SCE_NP_BASIC_EVENT_MESSAGE;
+							case SCE_NP_BASIC_MESSAGE_MAIN_TYPE_ADD_FRIEND:
```
Comment (Megamouse) [inline rpcs3/Emu/NP/np_handler.cpp:1119]: There's a missing break. Either add it or add fallthrough

File: rpcs3/Emu/NP/np_handler.cpp [hunk 2]
```diff
@@ -970,6 +970,11 @@ namespace np
 		}
 	}
 
+	void np_handler::send_message(const message_data& msg_data, const std::set<std::string>& npids)
+	{
+		rpcn->send_message(msg_data, npids);
```
Comment (Megamouse) [inline rpcs3/Emu/NP/np_handler.cpp:975]: if rpcn

File: rpcs3/Emu/NP/np_handler.cpp (hunk 1 above)
Comment (RipleyTom) [reply rpcs3/Emu/NP/np_handler.cpp:1119]: Isn't fallthrough implied if there is literally no code?

Comment (RipleyTom) [reply rpcs3/Emu/NP/np_handler.cpp:1120]: Oops nvm, I see what you meant :D

Comment (elad335) [reply rpcs3/Emu/NP/np_handler.cpp:1120]: it generates a warning on some compilers iirc

**Pull request merged**